File: ios/Classes/Data/RecommendationOptionsEncoder.swift
//
//  RecommendationOptionsEncoder.swift
//  exponea
//
//  Created by Franky on 29/06/2021.
//

import Foundation
import ExponeaSDK

class RecommendationOptionsEncoder {
    static func decode(_ data: [String:Any?]) throws -> RecommendationOptions {
        return RecommendationOptions.init(
            id: try data.getRequired("id"),
            fillWithRandom: try data.getRequired("fillWithRandom"),
            size: Int((try data.getOptional("size") as Double?) ?? 10.0),
            items: try data.getOptional("items"),
            noTrack: try data.getOptional("noTrack") ?? false,
            catalogAttributesWhitelist:  try data.getOptional("catalogAttributesWhitelist")
        )
    }
}

File: android/src/main/kotlin/com/exponea/data/RecommendationOptionsEncoder.kt
package com.exponea.data

import com.exponea.sdk.models.CustomerRecommendationOptions

class RecommendationOptionsEncoder {
    companion object {
        fun decode(data: Map<String, Any?>): CustomerRecommendationOptions {
            return CustomerRecommendationOptions(
                id = data.getRequired("id"),
                fillWithRandom = data.getRequired("fillWithRandom"),
                size = data.getOptional<Double>("size")?.toInt() ?: 10,
                items = data.getOptional("items"),
                noTrack = data.getOptional("noTrack"),
                catalogAttributesWhitelist = data.getOptional("catalogAttributesWhitelist")
            )
        }
    }
}